src/components/Courses.js

```diff
@@ -1,149 +1,128 @@
 import CourseHeader from './CourseHeader'
+import ErrorMessages from './ErrorMessages'
 import ExpandableCourse from './ExpandableCourse'
 import Instructors from './Instructors'
 import Meetings from './Meetings'
-import WaitlistCourse from './WaitlistCourse'
-import { fetch_courses } from './../actions/coursesActions'
-import { getBookButton } from './BuyBooks'
-import { useSelector, useDispatch } from 'react-redux'
-
-import Card from '@material-ui/core/Card'
-import CardActions from '@material-ui/core/CardActions'
-import CardContent from '@material-ui/core/CardContent'
 import Typography from '@material-ui/core/Typography'
-import { makeStyles } from '@material-ui/styles';
+import WaitlistCourse from './WaitlistCourse'
+import { makeStyles } from '@material-ui/styles'
+import { useSelector } from 'react-redux'
 
-const useStyles = makeStyles(theme => ({
+const useStyles = makeStyles((theme) => ({
   courseContainer: {
     flex: '1 1 auto',
-    padding: '1em'
+    padding: '1em',
   },
-
-  buttonsDiv: {
-    display: 'flex',
-    justifyContent: 'start',
-    alignItems: 'center'
+  card: {
+    backgroundColor: '#fafafa',
   },
-
-  button: {
-    margin: theme.spacing()
+  content: {
+    paddingTop: 0,
+    display: 'flex',
+    justifyContent: 'center',
   },
-
-  rightIcon: {
-    marginLeft: theme.spacing()
+  empty: {
+    textAlign: 'center',
   },
-
-  coursesDiv: {
+  loading: {
     display: 'flex',
-    flexFlow: 'wrap'
+    marginTop: 50,
+    justifyContent: 'center',
   },
-
-  coursesDivMobile: {
+  error: {
     display: 'flex',
-    flexDirection: 'column'
+    justifyContent: 'center',
   },
-
-  card: {
-    backgroundColor: '#fafafa'
+  container: {
+    marginTop: '1em',
   },
-
-  content: {
-    paddingTop: 0,
-    display: 'flex',
-    justifyContent: 'center'
+  actions: {
+    flexWrap: 'wrap',
+    justifyContent: 'center',
   },
-
-  empty: {
-    textAlign: 'center'
-  },
-
-  progress: {
-    margin: `0 ${theme.spacing(2)}px`
-  }
 }))
 
 export default function Courses(props) {
   const classes = useStyles()
-  const current_term = useSelector(state => state.terms.current_term)
-  const courses = useSelector(state => state.courses.courses)
-  const { mobile } = props
-  const dispatch = useDispatch();
-
-  useEffect(() => {
-    dispatch(fetch_courses(current_term))
-  }, [dispatch, current_term])
+  const courses = useSelector((state) => state.courses)
+  const courses_fetched = useSelector((state) => state.fetched)
+  const courses_fetching = useSelector((state) => state.fetching)
+  const courses_error = useSelector((state) => state.error)
 
-  if (courses === null || courses === []) {
+  if (courses_fetching === true) {
     return (
-      <Typography variant="h3" className={classes.empty} tabIndex="0">
+      <div className={classes.loading}>
+        <CircularProgress color='secondary' size={50} />
+      </div>
+    )
+  } else if (courses_fetched === true && courses_error === true) {
```

Comment:
Since these are boolean values you won't need to write out the full `=== true` each time. There are a few other places where these expressions could be shortened.

Reply:
True. Some of this code could have been copy/pasted from other areas or I was just writing it to match what was already there. But I can go through and shorten things where possible. Thanks!

Reply:
Going to address this in #212